### PR TITLE
Introduce a target picker for the logbook card

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Introduce a target picker on the logbook card. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/36411

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by MindFreeze** [comment]
We discussed this and it does make sense to have it on the card so it is aligned with the logbook panel. Thanks, @silamon

File: src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts [hunk 1]
```diff
@@ -60,8 +64,16 @@ export class HuiLogbookCardEditor
     this._config = config;
   }
 
-  get _entities(): string[] {
-    return this._config!.entities || [];
+  get _targetPicker(): HassServiceTarget {
+    const entities = this._config!.entities || [];
+    if (this._config!.entities) {
+      delete this._config!.entities;
+    }
```
Comment (wendevlin) [inline src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts:71]: Have you tested this?
I get `TypeError: Cannot delete property 'entities' of #<Object>`

File: src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts [hunk 2]
```diff
@@ -77,25 +89,18 @@ export class HuiLogbookCardEditor
...
-          "ui.panel.lovelace.editor.card.config.required"
-        )})`}
-      </h3>
-      <ha-entities-picker
+
+      <ha-target-picker
         .hass=${this.hass}
-        .value=${this._entities}
+        .value=${this._targetPicker}
         .entityFilter=${filterLogbookCompatibleEntities}
```
Comment (wendevlin) [inline src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts:96]:
```suggestion
        .entityFilter=${filterLogbookCompatibleEntities}
        addOnTop
```

Then it is like in logbook panel.

File: src/panels/lovelace/cards/hui-logbook-card.ts [hunk 3]
```diff
@@ -68,11 +110,33 @@ export class HuiLogbookCard extends LitElement implements LovelaceCard {
...
+      this.hass.devices,
+      this.hass.areas
     );
+    if (entities.length === 0) {
+      return undefined;
+    }
+    return entities;
   }
 
+  private __getEntityIds = memoizeOne(
```
Comment (wendevlin) [inline src/panels/lovelace/cards/hui-logbook-card.ts:130]:
```suggestion
  private _getMemoizedEntityIds = memoizeOne(
```

**Review by wendevlin — CHANGES_REQUESTED** [review]

File: src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts (hunk 1 above)
Comment (silamon) [reply src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts:71]: Definitely tested, back and forth. I'll take a look.

File: src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts [hunk 4]
```diff
@@ -77,25 +93,19 @@ export class HuiLogbookCardEditor
...
-        )})`}
-      </h3>
-      <ha-entities-picker
+
+      <ha-target-picker
         .hass=${this.hass}
-        .value=${this._entities}
         .entityFilter=${filterLogbookCompatibleEntities}
+        .value=${this._targetPicker}
+        addOnTop
```
Comment (wendevlin) [inline src/panels/lovelace/editor/config-elements/hui-logbook-card-editor.ts:101]:
```suggestion
        add-on-top
```

sorry the attribute is `add-on-top`

**Review by wendevlin — CHANGES_REQUESTED** [review]

**Review by wendevlin — APPROVED** [review]
Thank you very much!

**Pull request merged**